app/helpers/time_variance_helper.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+# Time Variance Helper
+# Time Variance Helper
+module TimeVarianceHelper
+  def self.adjust_prices_for_all_items(character)
+    setup_random_seed(character)
+    update_all_item_prices
+    reset_random_seed
+  end
+
+  def self.setup_random_seed(character)
+    current_day_seed = character.day
+    srand(current_day_seed)
+    puts 'Adjusting prices based on the current day seed'
+  end
+
+  def self.update_all_item_prices
+    Item.find_each do |item|
+      update_item_price(item)
```

Comment:
It is probably not a great idea to have the items in the database updated, as if multiple players are accessing this application then this will cause conflicting updates. The changes should only reflect in the data sent back to one specific player; i.e. applied in the controller producing the trading page/handling trade actions.